taskcluster/ac_taskgraph/transforms/build.py

```diff
@@ -31,21 +52,36 @@ def handle_coverage(config, tasks):
     for task in tasks:
         if task.pop('include-coverage'):
             task['run']['gradlew'].insert(0, '-Pcoverage')
+            task['run']['post-gradlew'] = [['automation/taskcluster/action/upload_coverage_report.sh']]
+            task['run']['secrets'] = [{
+                'name': 'project/mobile/android-components/public-tokens',
+                'key': 'codecov',
+                'path': '.cc_token',
+            }]
 
         yield task
 
 
 @transforms.add
-def format_component_name_and_lint(config, tasks):
-    for task in tasks:
-        lint_task = 'lintRelease' if task.pop("lint-release") else 'lint'
+def format_component_name_and_timestamp(config, tasks):
+    timestamp = _get_timestamp(config)
 
+    for task in tasks:
         for field in ('description', 'run.gradlew', 'treeherder.symbol'):
-            _deep_format(task, field, component=task["name"], lint_task=lint_task)
+            component = task["attributes"]["component"]
+            _deep_format(
+                task, field, component=component, timestamp=timestamp
+            )
 
         yield task
 
 
+def _get_timestamp(config):
+    push_date_string = config.params["moz_build_date"]
+    push_date_time = datetime.datetime.strptime(push_date_string, "%Y%m%d%H%M%S")
```

Comment:
I've been wondering the same. I ended up not putting it there, because the timestamp is just used in this transform **and** it's a derivative of `moz_build_date`. If it were used elsewhere, I would have placed it in `Parameters`. I don't mind exposing it later in a followup.

Reply:
I was suggesting adding a property to `Parameters` that exposes `moz_build_date` as a `datetime` instance, rather than adding a new parameter. Definitely fodder for a follow up.

Reply:
Got it! Something like this https://phabricator.services.mozilla.com/D48502 ?